sdk/trace/provider.go

```diff
@@ -350,6 +354,25 @@ func WithSpanLimits(sl SpanLimits) TracerProviderOption {
 	})
 }
 
+func applyTracerProviderEnvConfigs(cfg *tracerProviderConfig) {
+	for _, opt := range tracerProviderOptionsFromEnv() {
+		opt.apply(cfg)
+	}
+}
```

Comment:
What do we gain by going through options instead of just directly applying to the config here?  I would suggest we just have a single function that is `tracerProviderConfigFromEnv() *tracerProviderConfig`.

Reply:
My idea here was to avoid depending on `tracerProviderConfig`, an unexported struct that could later potentially be exported publically. By only using the public configuration interfaces provided by `sdk/trace`, we avoid breaking something like `tracerProviderConfigFromEnv` if `tracerProviderConfig` was to change in the future (although, fixing that shouldn't be that tough, so it's not really that great of a reason :smiley: ). This is the pattern I based it on: https://github.com/open-telemetry/opentelemetry-go/blob/aedef05e953cbca349f1d8b4dfd46f69d96c6a61/exporters/otlp/otlptrace/otlptracehttp/client.go#L81

Reply:
That being said, `otlpconfig` is an external package, which doesn't have access to the configuration struct, unlike here. Nonetheless, I still thought that only depending on exported configuration interfaces keeps things "idiomatic" and less prone to breakage. What do you think?